haystack/document_store/sql.py

```diff
@@ -440,6 +440,23 @@ def query_by_embedding(self,
                                   "Change the query type (e.g. by choosing a different retriever) "
                                   "or change the DocumentStore (e.g. to ElasticsearchDocumentStore)")
 
+    def delete_document_by_id(self, id: Union[List[str], str], index: Optional[str] = None):
+        """
+        Delete document given its id
+        :param id: Unique document identifier
+        :param index: Index name to delete the document from.
+        """
+        try:
+            document_to_delete = self.session.query(DocumentORM.id).filter_by(id, index=index)
+            if not id:
+                raise Exception("Document could not be found")
+            self.session.query(DocumentORM).filter(DocumentORM.id.in_(document_to_delete)).delete(synchronize_session=False)
```

Comment:
We can simply call 
```
document_to_delete.delete(synchronize_session=False)
```

Reply:
Also this statement can have synchronisation issue, so better to run everything in one go with `synchronize_session=True`. WDYT?

Reply:
Per SQLAlchemy documentation, it is bad practice to synchronise the session inside the function that handles database transaction.
When a commit() method is executed, the Session object will expire all the states. Therefore, all new states will be reloaded for another transaction (read, write,etc)
As the session is not kept that long, I'd leave it like it is

Reply:
Yes you are correct let's not use `synchronize_session=True`